### PR TITLE
Release notes for API Designer 2.5.0

### PR DESCRIPTION
**Pull request by rddheath**

### CODE REVIEW

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 1]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+* When you are using Microsoft Internet Explorer or Microsoft Edge as your browser and you click *Design Center* or the hamburger menu in the top-left corner as you are using the code editor, there is no longer a browser pop-up asking whether you want to leave the page.
+// DESIGNER-3710
+* It is now possible to view inline documentation for errors if you are using Microsoft Internet Explorer, Microsoft Edge, or Mozilla Firefox as your browser. Click the information icon that appears next to error messages.
+
+=== Fixed Issues
+
+// SE-12002
+* Changes made in the code editor might take up to six minutes to be parsed.
+// SE-11942
+* In the code editor, creating a fragment in an API-specification project, leaving the fragment without content, and then adding a reference to it from the main file before adding content to the fragment, caused the reference to be parsed as an error
```
Comment (mcherem) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Period at the end of the sentence?

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 2]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
++
+.Example
+----
+#%RAML 1.0 NamedExample
+"customerProfile": {
+  "password": "abc123"
+}
+----
+// SE-11617
+* For certain API specifications, the API console was not appearing.
```
Comment (mcherem) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Do we use 'displaying' or 'appearing'?

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 3]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+== API Designer 2.5.0
+*June 15, 2019*
+
+This release introduces several new features in the visual editor.
+
+=== New Features in the Code Editor
+// DESIGNER-3615
+* You can now import binary files, such as PDF and PNG files, into your projects. Such files are not parsed and no preview for them is available.
+// MSERV-285
+* Use behavioral headers with the mocking service to modify in various ways how the mocking service behaves. See xref:design-center::apid-behavioral-headers.adoc[Add Behavioral Headers to Simulated API Calls].
```
Comment (mcherem) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: I wanted to see, 'in various ways' in parenthesis.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 4]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+
+=== New Features in the Visual Editor
+
+* Create security schemes for your API specifications. The different supported types are OAuth 1.0, OAuth 2.0, Basic Authentication, Digest Authentication, and Pass Through.
+* Define union types for elements.
++
+In previous releases, you could select only one data type at a time for properties, bodies, payloads, headers, and any other kind of element. Now, you can choose multiple data types to create _union_ data types for any of those elements.
++
+The chosen data types must have the same built-in type. For example, if you choose two data types to create a union data type from, both data types must be strings, or both must be numbers, and so on. You must not create a union data type from a custom data type that is a string and another that is an object. If you do, you cannot edit other properties of the element that you have created the union datatype for.
```
Comment (iamerico) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Define union types for elements.
This behavior was modified, it should be something like:

> In previous releases, you could select only one data type at a time for properties, bodies, payloads, headers, and any other kind of element. Now, you can choose multiple data types to create _union_ data types for any of those elements.
> 
> When creating a union type, the user won't be able to edit any custom properties of the union type itself. An instance of the datatype should meet all restrictions of at least one of the super types.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 5]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+=== Fixed Issues
+
+// SE-12002
+* Changes made in the code editor might take up to six minutes to be parsed.
+// SE-11942
+* In the code editor, creating a fragment in an API-specification project, leaving the fragment without content, and then adding a reference to it from the main file before adding content to the fragment, caused the reference to be parsed as an error
+// SE-11944
+* Underscores, spaces, and hyphen were not parsed as separator characters in variables in resourceTypes.
+// SE-11832
+* It was possible to create API specifications without the appropriate entitlement.
```
Comment (iamerico) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: SE-11832
This is a major change, you could add:
> It was possible to create API specifications without the appropriate entitlement.
> 
> A new authorization layer has been added to every API Designer xAPI endpoint. This authorization consists on having the Design Center API entitlement enabled for the requesting user. Hence, with this release, every API Designer xAPI or Anypoint CLI user will require the Designer Center API entitlement to perform any request to our API (in the same way Anypoint Platform users need the entitlement to create or access an API Designer project).

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 6]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+* For certain API specifications, the API console was not appearing.
+// SE-11318
+* Not all types were being displayed in the list of types in the API console.
+// SE-11208
+* When a specification defines a boolean value but does not provide a default value, the API console now provides a default value. Doing so prevents problems from arising with the mocking service.
+// SE-11123
+* In some cases, an API-specification project might never finish loading when an attempt was made to open it.
+// SE-10576
+* API specifications that included binary files were not displayed properly in Exchange after being published.
+// SE-11925
```
Comment (iamerico) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: SE-11925
This should be in Release Notes of Mocking Service v2.2.0

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 7]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+* The mocking service included invalid tags in returned XML when the data type contained an array.
+// SE-11683
+* The mocking service converted null values to string "null" values.
```
Comment (iamerico) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: SE-11683
This should be in Release Notes of Mocking Service v2.2.0

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 3 above)
Comment (iamerico) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: MSERV-285
This should be in Release Notes of Mocking Service v2.2.0

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc [hunk 8]
```diff
@@ -19,6 +20,108 @@ These release notes cover the following versions of these features:
...
+// SE-11788
+
```
Comment (iamerico) [inline modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: SE-11788
In some cases, references could not be resolved when examples were included. The error message was confusing and with little information about the cause.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 1 above)
Comment (rddheath) [reply modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Fixed.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 2 above)
Comment (rddheath) [reply modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: "Display" is a transitive verb, while "appear" is intransitive. So, the correct choice here is the latter.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 3 above)
Comment (rddheath) [reply modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Nope, "in various ways" is not a parenthetical comment.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 6 above)
Comment (rddheath) [reply modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Fixed.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 7 above)
Comment (rddheath) [reply modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: Fixed.

File: modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc (hunk 3 above)
Comment (rddheath) [reply modules/ROOT/pages/design-center/design-center-release-notes-api_specs.adoc]: I copied it into the release notes for the mocking service. I think a copy of this note should remain here for APID users.

File: modules/ROOT/pages/api-mocking-service/api-mocking-service-release-notes.adoc [hunk 9]
```diff
@@ -17,6 +18,21 @@ endif::[]
 * <<API Mocking Service 1.1.2>>
 * <<API Mocking Service 1.1.1>>
 
+== API Mocking Service 2.2.0
```
Comment (iamerico) [inline modules/ROOT/pages/api-mocking-service/api-mocking-service-release-notes.adoc]: You could add one more items for the Mocking Service RN:
> MSERV-600:
> Now the Mock Service supports XML serialization facets such as "wrapped", "attribute" and "name" when generating an XML response based on a RAML example.

**Pull request merged**